airbyte-integrations/connectors/source-chargebee/source_chargebee/source.py

```diff
@@ -0,0 +1,205 @@
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import pendulum
+import requests
+import backoff
+
+# Chargebee
+import chargebee
+from chargebee.main import ChargeBee
+from chargebee.list_result import ListResult  # stores next_offset
+from chargebee.models import Subscription, Customer, Invoice, Order, Plan, Addon
+from chargebee.api_error import OperationFailedError
+
+# Airbyte
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.models import AirbyteStream, SyncMode
+from airbyte_cdk.logger import AirbyteLogger
+
+# Backoff params below
+# according to Chargebee's guidance on rate limit
+# https://apidocs.chargebee.com/docs/api?prod_cat_ver=2#api_rate_limits
+MAX_TRIES = 10  # arbitrary max_tries
+MAX_TIME = 90  # because Chargebee API enforce a per-minute limit
+
+
+class ChargebeeStream(Stream):
```

Comment:
Would this be easier if you use the Chargebee HTTP API and this way you can use a vanilla `HttpStream` instead? Either way, nicely done writing this stream from scratch!